### PR TITLE
Hosted improvement

### PR DESCRIPTION
**Pull request by Sidddddarth**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by chandumlg — APPROVED** [review]

**Review by psrikanth88 — APPROVED** [review]

**Review by snarkychef — CHANGES_REQUESTED** [review]

File: jobsdb/jobsdb.go [hunk 1]
```diff
@@ -532,7 +542,13 @@ func loadConfig() {
 	config.RegisterDurationConfigVariable(time.Duration(5), &refreshDSListLoopSleepDuration, true, time.Second, "JobsDB.refreshDSListLoopSleepDurationInS")
 	config.RegisterDurationConfigVariable(time.Duration(5), &backupCheckSleepDuration, true, time.Second, "JobsDB.backupCheckSleepDurationIns")
 	useJoinForUnprocessed = config.GetBool("JobsDB.useJoinForUnprocessed", true)
-
+	config.RegisterBoolConfigVariable(true, &useNewCacheBurst, true, "JobsDB.useNewCacheBurst")
+	config.RegisterIntConfigVariable(1, &maxWriters, false, 1, "JobsDB.maxWriters")
```
[the remark this answers is not in the document]
Comment (snarkychef) [reply jobsdb/jobsdb.go:549]: Handled in a separate PR

File: jobsdb/jobsdb.go [hunk 2]
```diff
@@ -186,9 +187,6 @@ func (jd *HandleT) StoreInTxn(txHandler *sql.Tx, jobList []*JobT) error {
 		return err
 	}
 
-	//Empty customValFilters means we want to clear for all
```
[the remark this answers is not in the document]
Comment (snarkychef) [reply jobsdb/jobsdb.go:189]: Handled in a separate PR

File: jobsdb/jobsdb.go [hunk 3]
```diff
@@ -490,6 +495,14 @@ var (
 	useJoinForUnprocessed                        bool
 	backupRowsBatchSize                          int64
 	pkgLogger                                    logger.LoggerI
+	maxReaders                                   int
+	maxWriters                                   int
+	storeChannelBufferLength                     int
+	storeWithRetryChannelBufferLength            int
```
[the remark this answers is not in the document]
Comment (snarkychef) [reply jobsdb/jobsdb.go:501]: Handled in a separate PR

File: jobsdb/jobsdb.go [hunk 4]
```diff
@@ -586,6 +606,16 @@ func (jd *HandleT) Setup(ownerType OwnerType, clearAll bool, tablePrefix string,
 	jd.statNewDSPeriod = stats.NewStat(fmt.Sprintf("jobsdb.%s_new_ds_period", jd.tablePrefix), stats.TimerType)
 	jd.statDropDSPeriod = stats.NewStat(fmt.Sprintf("jobsdb.%s_drop_ds_period", jd.tablePrefix), stats.TimerType)
 
+	config.RegisterBoolConfigVariable(true, &jd.enableWriterQueue, true, fmt.Sprintf("JobsDB.%v.enableWriterQueue", jd.tablePrefix))
```
[the remark this answers is not in the document]
Comment (snarkychef) [reply jobsdb/jobsdb.go:609]: Handled in a separate PR

File: jobsdb/jobsdb.go [hunk 5]
```diff
@@ -676,6 +706,71 @@ func (jd *HandleT) readerWriterSetup() {
...
+		} else if readReq.reqType == Throttled.State {
+			readReq.jobsListChan <- jd.getThrottled(readReq.getQueryParams)
+		} else if readReq.reqType == Waiting.State {
+			readReq.jobsListChan <- jd.getWaiting(readReq.getQueryParams)
+		} else if readReq.reqType == NotProcessed.State {
+			readReq.jobsListChan <- jd.getUnprocessed(readReq.getQueryParams)
+		} else if readReq.reqType == Executing.State {
+			readReq.jobsListChan <- jd.getExecuting(readReq.getQueryParams)
+		} else {
+			panic(fmt.Errorf("unknown read request type: %s", readReq.reqType))
```
[the remark this answers is not in the document]
Comment (snarkychef) [reply jobsdb/jobsdb.go:769]: Handled in a separate PR

File: jobsdb/jobsdb.go [hunk 6]
```diff
@@ -1271,6 +1366,9 @@ func (jd *HandleT) dropDS(ds dataSetT, allowMissing bool) {
 	err = txn.Commit()
 	jd.assertError(err)
 
+	//Bursting Cache for this dataset
+	jd.markClearEmptyResult(ds, []string{}, []string{}, nil, hasJobs, nil)
```
[the remark this answers is not in the document]
Comment (snarkychef) [reply jobsdb/jobsdb.go:1370]: Handled in a separate PR

**Pull request merged**